compute_endpoint/tests/integration/endpoint/executors/test_gcengine_retries.py

```diff
@@ -0,0 +1,149 @@
+import logging
+
+# import tempfile
+import uuid
+from queue import Queue
+
+import pytest
+from globus_compute_common import messagepack
+from globus_compute_endpoint.engines import GlobusComputeEngine
+from globus_compute_endpoint.strategies import SimpleStrategy
+from globus_compute_sdk.serialize import ComputeSerializer
+from parsl.executors.high_throughput.interchange import ManagerLost
+from parsl.providers import LocalProvider
+from tests.utils import double, ez_pack_function, kill_manager, succeed_after_n_runs
+
+
+@pytest.fixture
+def gc_engine_with_retries(tmp_path):
+    ep_id = uuid.uuid4()
+    engine = GlobusComputeEngine(
+        address="127.0.0.1",
+        heartbeat_period=1,
+        heartbeat_threshold=1,
+        max_retries_on_system_failure=2,
+        provider=LocalProvider(
+            init_blocks=0,
+            min_blocks=0,
+            max_blocks=1,
+        ),
+        strategy=SimpleStrategy(interval=0.1, max_idletime=0),
+    )
+    queue = Queue()
+    engine.start(endpoint_id=ep_id, run_dir=str(tmp_path), results_passthrough=queue)
+
+    yield engine
+    engine.shutdown()
+
+
+def test_gcengine(gc_engine_with_retries):
+    engine = gc_engine_with_retries
+    task_id = uuid.uuid1()
+    serializer = ComputeSerializer()
+    task_body = ez_pack_function(serializer, double, (3,), {})
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(
+            task_id=task_id, container_id=uuid.uuid1(), task_buffer=task_body
+        )
+    )
+    future = engine.submit(task_id, task_message)
+
+    # Confirm that there are no retries on a normal function
+    packed_result = future.result()
+    result = messagepack.unpack(packed_result)
+    assert result.task_id == task_id
+    assert not result.error_details
+
+    # Confirm error message for ManagerLost
+    task_id = uuid.uuid1()
+    task_body = ez_pack_function(serializer, kill_manager, (), {})
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(
+            task_id=task_id, container_id=uuid.uuid1(), task_buffer=task_body
+        )
+    )
+
+    future = engine.submit(task_id, task_message)
+
+    try:
+        packed_result = future.result()
+    except ManagerLost as e:
+        logging.warning(f"This is the exception : {packed_result}")
+        logging.warning(f"Caught result exception : {packed_result}")
+        logging.warning(f"Exception body : {e}")
+    logging.warning("DEBUG: 2")
+
+    engine.shutdown()
+
```

Comment:
This `.shutdown()` call is unnecessary, as the fixture takes care of this, no?

Reply:
I just want to make it super obvious, because if it isn't cleaned up, it can affect the following tests.